airflow/contrib/operators/gcp_container_operator.py

```diff
@@ -170,3 +177,141 @@ def execute(self, context):
+    template_fields = ('project_id', 'location',
+                       'cluster_name') + KubernetesPodOperator.template_fields
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 location,
+                 cluster_name,
+                 gcp_conn_id='google_cloud_default',
+                 key_file=None,
+                 *args,
+                 **kwargs):
+        """
+        Executes a task in a Kubernetes pod in the specified Google Kubernetes Engine
+        cluster
+
+        This Operator assumes that the system has gcloud installed and has default
+        application credentials set in order to use gcloud with the corresponding project,
+        cluster, etc.
+
+        The **minimum** required to define a cluster to create are the variables
+        ``task_id``, ``project_id``, ``location``, ``cluster_name``, ``name``,
+        ``namespace``, and ``image``
+
+        **Operator Creation**: ::
+
+            operator = GKEPodOperator(task_id='pod_op',
+                                      project_id='my-project',
+                                      location='us-central1-a',
+                                      cluster_name='my-cluster-name',
+                                      name='task-name',
+                                      namespace='default',
+                                      image='perl')
+
+        .. seealso::
+            For more detail about application authentication have a look at the reference:
+            https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application
+
+        This does not require a gcp_conn_id nor api_version since it does not end up
+        getting a connection from a hook, nor using a specific api version
+
+        :param project_id: The Google Developers Console project id
+        :type project_id: str
+        :param location: The name of the Google Kubernetes Engine zone in which the
+            cluster resides, e.g. 'us-central1-a'
+        :type location: str
+        :param cluster_name: The name of the Google Kubernetes Engine cluster the pod
+            should be spawned in
+        :type cluster_name: str
+        """
+        super(GKEPodOperator, self).__init__(*args, **kwargs)
+        self.project_id = project_id
+        self.location = location
+        self.cluster_name = cluster_name
+        self.gcp_conn_id = gcp_conn_id
+        self.key_file = key_file
+
+    def execute(self, context):
+        # Specifying a service account file allows the user to using non default
+        # authentication for creating a Kubernetes Pod. This is done by setting the
+        # environment variable `GOOGLE_APPLICATION_CREDENTIALS` that gcloud looks at.
+        # If not specified it will use the default service account that GKE uses.
+        if self.key_file:
+            if os.path.isfile(self.key_file):
+                os.environ[G_APP_CRED] = self.key_file
```

Comment:
Shouldn't the key_file be specified in the GCP connection?

Reply:
Yes, it is, at this point it is set up to allow users to specify either a gcp_conn_id which has the path to key file or json, OR   a plain path to a key file. The plain path takes precedence.   Let me know if you'd like to only support specifying key file through gcp_conn_id